src/main/java/com/juniorstart/juniorstart/service/UserProfileService.java

```diff
@@ -0,0 +1,125 @@
+        List<String> convertedTechnology = technology.stream().map(WordUtils::capitalize).collect(Collectors.toList());
+        ValidatePageUtil.validatePageNumberAndSize(page, size);
+
+        Pageable pageable = PageRequest.of(page, size);
+        Page<UserProfile> foundUsers = userProfileRepository.findByUserTechnology_technologyNameInAndUserRoleIn(convertedTechnology, userRole, pageable);
+
+        return getUserProfilePagedResponse(foundUsers);
+    }
+
+    private PagedResponse<UserProfile> getUserProfilePagedResponse(Page<UserProfile> foundUsers) {
+        if(foundUsers.getNumberOfElements() == 0) {
+            return new PagedResponse<>(Collections.emptyList(), foundUsers.getNumber(), foundUsers.getSize(), foundUsers.getTotalElements(), foundUsers.getTotalPages(), foundUsers.isLast());
+        }
+
+        return new PagedResponse<>(foundUsers.toList(), foundUsers.getNumber(), foundUsers.getSize(), foundUsers.getTotalElements(), foundUsers.getTotalPages(), foundUsers.isLast());
+    }
+
+    /** Get a List of UserProfile.
+     * @param technology Technology name you are looking for
+     * @return list of UserProfile
+     */
+    public PagedResponse<UserProfile> findByTechnology(List<String> technology, int page, int size) {
+        List<String> convertedTechnology = technology.stream().map(WordUtils::capitalize).collect(Collectors.toList());
+        ValidatePageUtil.validatePageNumberAndSize(page, size);
+
+        Pageable pageable = PageRequest.of(page, size);
+        Page<UserProfile> foundUsers = userProfileRepository.findByUserTechnology_technologyNameIn(convertedTechnology, pageable);
+        return new PagedResponse<>(foundUsers.toList(), foundUsers.getNumber(), foundUsers.getSize(), foundUsers.getTotalElements(), foundUsers.getTotalPages(), foundUsers.isLast());
+        //return getUserProfilePagedResponse(foundUsers);
+    }
+
+    /** Get a List of UserProfile.
+     * @param userRole UserRole(JUNIOR,MENTOR etc) name you are looking for
+     * @return list of UserProfile
+     * @throws ResourceNotFoundException if userRole isn't valid
+     */
+    public PagedResponse<UserProfile> findByUserRole(List<UserRole> userRole, int page, int size) {
+        ValidatePageUtil.validatePageNumberAndSize(page, size);
+
+        Pageable pageable = PageRequest.of(page, size);
+        Page<UserProfile> foundUsers = userProfileRepository.findByUserRoleIn(userRole, pageable);
+
+        return getUserProfilePagedResponse(foundUsers);
+    }
+    /** Validate Listof String and return ENUM values of UserRole.
+     * @param userRole UserRole(JUNIOR,MENTOR etc) name you are looking for
+     * @return list of UserProfile
+     * @throws ResourceNotFoundException if userRole isn't valid
+     */
+    public List<UserRole> validateAndReturnAsEnum(List<String> userRole) {
+        for (String value : userRole) {
```

Comment:
Można tutaj użyć stream, ale to tylko moja sugestia

Reply:
Hmm, czyli mam zamieniać ? Bo nie do końca wiem, czemu tutaj np byłby Stream lepszy

Reply:
moim zdaniem bardziej przejrzyście byłoby napisane